src/schemas.ts

```diff
@@ -20,6 +28,16 @@ export const SignupFormSchema = yup.object().shape({
         .matches(/^(?=.*[0-9]).{8,}$/, "Entry must contain at least one number")
         .matches(/^(?=.*[!@#$%^&*(),.?\":{}|<>]).{8,}$/, "Entry must contain at least one special character")
         .matches(/^(?!.* ).{8,}$$/, "Entry must NOT contain a space"),
```

Comment:
The validation on "no space" can be taken out. It may be a bit oblivious to users. You can just apply a .trim() to remove trailing spaces.

Reply:
Wouldn't this allow spaces in the password, as `trim()` only addresses space at the beginning and end?